manifests/unit.pp

```diff
@@ -22,55 +22,69 @@
   $install_wantedby        = undef,
 ) {
 
-  validate_re($ensure, [ '^present$', '^absent$' ],
-    "systemd::unit::${name}::ensure is invalid and does not match the regex.")
+  include ::systemd
+
+  if is_string($ensure) {
+    validate_re($ensure, [ '^present$', '^absent$' ],
+      "systemd::unit::${name}::ensure is not a string or is invalid and does not match the regex.")
```

Comment:
isn't a string if it's in that condition?

Reply:
Hm. Ah yes. It should be "is a string but does not match the regex" or something.

Reply:
or just like the message before